core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -131,8 +131,11 @@ class LogManager(logDirs: Array[File],
 
     val liveLogDirs = new ConcurrentLinkedQueue[File]()
 
-    for (dir <- dirs if !initialOfflineDirs.contains(dir)) {
+    for (dir <- dirs) {
       try {
+        if (initialOfflineDirs.contains(dir))
+          throw new IOException(s"${dir.getAbsolutePath} fails to be loaded during broker startup")
```

Comment:
Nit: s"Failed to load ${dir.getAbsolutePath} during broker startup." would be more consistent with the other error messages.

Reply:
Good point. I will update the patch as suggested.